app/models/host.rb

```diff
@@ -639,10 +639,13 @@ def verify_credentials_task(userid, auth_type = nil, options = {})
     MiqTask.generic_action_with_callback(task_opts, queue_opts)
   end
 
-  def verify_credentials?(*args)
+  def verify_credentials?(auth_type = nil, options = {})
     # Prevent the connection details, including the password, from being leaked into the logs
     # and MiqQueue by only returning true/false
-    !!verify_credentials(*args)
+    auth = options.delete(:credentials)
+    update_authentication(auth, :save => false) if auth.present?
```

Comment:
Hm this doesn't looks right, if we're verifying new credentials for an existing host and the verification fails then we've just overwritten the old credentials right?

Why did you have to add this?  Or was this already done elsewhere that you had to move?

Reply:
This is the way it has always been done (not a fan)

We set the credentials, we pass in `:save => false` so they are not changed
Then we call verify on the host.
It uses the temporary values.